Sources/Citadel/Algorithms/ED25519.swift

```diff
@@ -0,0 +1,98 @@
+import NIO
+import NIOFoundationCompat
+import BigInt
+import NIOSSH
+import CCryptoBoringSSL
+import Foundation
+import Crypto
+
+public enum ED25519 {
+    public final class PublicKey: NIOSSHPublicKeyProtocol {
```

Comment:
Is it not possible to conform `Curve25519.Signing.PublicKey` to this protocol?

Reply:
I tried that at first, not sure where I got stuck. I'll give it another go real quick now that the implementations is more complete.

Reply:
Unfortunately we run into problems when trying to conform `Curve25519.Signing.PublicKey` to NIOSSHPrivateKeyProtocol since it requires a property of `var publicKey: NIOSSHPublicKey` but the private key object already has a `var publicKey: PublicKey` Swift unfortunately doesn't recognise this as a valid implementation of the protocol, but I can't access it from the property itself due to (I think) the names clashing?